frontends/api/src/hooks/learningResources/index.ts

```diff
@@ -298,13 +294,21 @@ const useUserListRelationshipCreate = () => {
         UserListRelationshipRequest: params,
       }),
     onSuccess: (response, _vars) => {
-      queryClient.setQueryData(
-        learningResources.detail(response.data.child).queryKey,
-        response.data.resource,
+      queryClient.setQueriesData<PaginatedLearningResourceList>(
+        learningResources.featured({}).queryKey,
+        (old) => updateListParentsOnAdd(response.data, old),
```

Comment:
Regarding `updateListParentsOnAdd` and `updateListParentsOnDestroy `:

1. The featured carousel on the homepage is populated by an API that has an intentionally randomized order.
2. In general throughout the app, when data might change, we tell react query "invalidate the cached data" and it re-fetches.
3. Adding the featured courses to a userlist changes their data (it affects `resource.user_list_parents). **However**, if we invalidate the cache and refetch new, updated data, then the carousel order will change, which is strange.
    - you can see this behavior on RC / Prod, but the bookmarks won't fill / unfill, so it's less noticeable that position has changed.
4. So instead of invalidating the cache and refetching the data, we manually update the data on the frontend. 


We **only** do this for userlists, not for learnignpaths, because...
- changing a resource's learningpath membershould can change whether it is a featured course or not, so we really do need to refetch the data.